test/extended/util/framework.go

```diff
@@ -189,7 +189,9 @@ var CheckImageStreamTagNotFoundFn = func(i *imageapi.ImageStream) bool {
 func WaitForADeployment(client kclient.ReplicationControllerInterface,
 	name string,
 	isOK, isFailed func(*kapi.ReplicationController) bool) error {
-	for {
+	startTime := time.Now()
+	endTime := startTime.Add(3 * time.Minute)
```

Comment:
bump this up to something like 15 or 20 minutes.  we don't want to introduce flakes if we have existing extended tests that take a long time waiting for a deployment for some reason.

i'm also surprised 3 minutes is long enough for you test, given that it basically has to get through the entire imagebuild+deploy step in 3 minutes to pass. 

(3 minutes or the watch timeout, whichever is longer.)